src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -2339,6 +2343,11 @@ void InterpreterSelectQuery::executeFetchColumns(QueryProcessingStage::Enum proc
         /// need replay operation
         if (settings.replay_speed > 0)
         {
+            if (settings.enable_backfill_from_historical_store.value)
```

Comment:
since backfill flag is true by default, let's reset it to `false` if `replay_speed=1` for example, otherwise user will need do `select * from my_stream settings replay_speed = 1, enable_backfill_from_historical_store=false`